### PR TITLE
Option to auto select radio station of Best Alternate B>

### PR DESCRIPTION
**Pull request by AlphaLima**
Option to auto select radio station of Best Alternate B>
in radio dialog for active and Standby frequency separately

image

### CODE REVIEW

File: Common/Source/Comm/device.cpp [hunk 1]
```diff
@@ -1224,27 +1225,38 @@ BOOL devPutFreqSwap() {
...
-    RadioPara.ActiveFrequency=  Freq;
-    _sntprintf( RadioPara.ActiveName, NAME_SIZE,_T("%s") , StationName);
-    return TRUE;
-  }
+if( ValidFrequency(Freq))
+{
+
+  RadioPara.ActiveFrequency=  Freq;
+  CopyTruncateString( RadioPara.ActiveName, NAME_SIZE, StationName);
+//  _sntprintf( RadioPara.ActiveName, NAME_SIZE,_T("%s") , StationName);
```
Comment (brunotl) [inline Common/Source/Comm/device.cpp:1237]: useless comments

File: Common/Source/Comm/device.cpp [hunk 2]
```diff
@@ -1224,27 +1225,38 @@ BOOL devPutFreqSwap() {
...
 BOOL devPutFreqStandby(double Freq, const TCHAR* StationName) {
-  if (SIMMODE) {
-     RadioPara.PassiveFrequency=  Freq;
-     _sntprintf( RadioPara.PassiveName, NAME_SIZE,_T("%s") , StationName);
-    return TRUE;
-  }
+if( ValidFrequency(Freq))
+{
+	RadioPara.PassiveFrequency=  Freq;
+//	_sntprintf( RadioPara.PassiveName, NAME_SIZE,_T("%s") , StationName);
```
Comment (brunotl) [inline Common/Source/Comm/device.cpp:1253]: useless comments

File: Common/Source/Dialogs/dlgRadioSettings.cpp [hunk 3]
```diff
@@ -154,48 +157,56 @@ int Idx=0;
   {
     RadioPara.Changed =FALSE;
     TCHAR Name[250];
-    if(_tcscmp(_T("        "), RadioPara.ActiveName ) == 0)
+    if((_tcscmp(_T("        "), RadioPara.ActiveName ) == 0)  || ( _tcslen(RadioPara.ActiveName) == 0))
```
Comment (brunotl) [inline Common/Source/Dialogs/dlgRadioSettings.cpp:160]: why compare with ("        ") ?

File: Common/Source/Dialogs/dlgRadioSettings.cpp [hunk 4]
```diff
@@ -154,48 +157,56 @@ int Idx=0;
...
       Idx = SearchStation(RadioPara.ActiveFrequency);
     if(Idx !=0)
     {
-        _stprintf(RadioPara.ActiveName,_T("%s"),WayPointList[Idx].Name);
+
+    		CopyTruncateString(RadioPara.ActiveName, NAME_SIZE, WayPointList[Idx].Name);
```
Comment (brunotl) [inline Common/Source/Dialogs/dlgRadioSettings.cpp:165]: can be removed ?

RadioPara.ActiveName will be overwrited by devPutFreqActive call line 170

File: Common/Source/Dialogs/dlgRadioSettings.cpp [hunk 5]
```diff
@@ -154,48 +157,56 @@ int Idx=0;
...
     if(wpnewActive)
       wpnewActive->SetCaption(Name);
     _stprintf(Name,_T("%6.03f"),RadioPara.ActiveFrequency);
     if(wpnewActiveFreq)
       wpnewActiveFreq->SetCaption(Name);
 
 
-    if(_tcscmp(_T("        "), RadioPara.PassiveName ) == 0)
-    Idx = SearchStation(RadioPara.PassiveFrequency);
+    if((_tcscmp(_T("        "), RadioPara.PassiveName ) == 0) ||( _tcslen(RadioPara.PassiveName) == 0))
```
Comment (brunotl) [inline Common/Source/Dialogs/dlgRadioSettings.cpp:190]: same as line 160

File: Common/Source/Dialogs/dlgRadioSettings.cpp [hunk 6]
```diff
@@ -154,48 +157,56 @@ int Idx=0;
...
+      Idx = SearchStation(RadioPara.PassiveFrequency);
     if(Idx !=0)
     {
-        _stprintf(RadioPara.PassiveName,_T("%s"),WayPointList[Idx].Name);
+
+        CopyTruncateString(RadioPara.PassiveName, NAME_SIZE, WayPointList[Idx].Name);
```
Comment (brunotl) [inline Common/Source/Dialogs/dlgRadioSettings.cpp:195]: same as line 165

File: Common/Source/Dialogs/dlgRadioSettings.cpp [hunk 7]
```diff
@@ -341,7 +353,7 @@ static void OnActiveButton(WndButton* pWnd){
 	return;
       }
       devPutFreqActive(Frequency, WayPointList[res].Name);
-      _stprintf(RadioPara.ActiveName,_T("%s"), WayPointList[res].Name);
+    	CopyTruncateString(RadioPara.ActiveName, NAME_SIZE, WayPointList[res].Name);
```
Comment (brunotl) [inline Common/Source/Dialogs/dlgRadioSettings.cpp:356]: already done by devPutFreqActive

File: Common/Source/Dialogs/dlgRadioSettings.cpp [hunk 8]
```diff
@@ -258,11 +270,11 @@ TCHAR Name[250];
     }
 
 
-    if((ValidWayPoint(PassiveRadioIndex)) && 0)
+    if((ValidWayPoint(PassiveRadioIndex)) )
     {
         if( PassiveRadioIndex > RESWP_END )
         {
-          _stprintf(Name,_T("%s"),WayPointList[PassiveRadioIndex].Name);
+        	CopyTruncateString(Name, DEVICE_NAME_LEN, WayPointList[PassiveRadioIndex].Name);
```
Comment (brunotl) [inline Common/Source/Dialogs/dlgRadioSettings.cpp:277]: why do a copy before setCaption ?

File: Common/Source/Dialogs/dlgRadioSettings.cpp [hunk 9]
```diff
@@ -272,7 +284,7 @@ TCHAR Name[250];
     }
     else
     {
-        _stprintf(Name,_T("%s"),RadioPara.PassiveName);
+      	CopyTruncateString(Name, DEVICE_NAME_LEN, RadioPara.PassiveName);
```
Comment (brunotl) [inline Common/Source/Dialogs/dlgRadioSettings.cpp:287]: same as line 277

File: Common/Source/Dialogs/dlgRadioSettings.cpp [hunk 10]
```diff
@@ -369,7 +381,8 @@ static void OnPassiveButton(WndButton* pWnd){
       }
       devPutFreqStandby(Frequency, WayPointList[res].Name);
 
-      _stprintf(RadioPara.PassiveName,_T("%s"), WayPointList[res].Name);
+  //    _stprintf(RadioPara.PassiveName,_T("%s"), WayPointList[res].Name);
+    	CopyTruncateString(RadioPara.PassiveName, DEVICE_NAME_LEN, WayPointList[res].Name);
```
Comment (brunotl) [inline Common/Source/Dialogs/dlgRadioSettings.cpp:385]: already done by devPutFreqStandby line 382

File: Common/Source/Dialogs/dlgRadioSettings.cpp [hunk 11]
```diff
@@ -394,16 +407,14 @@ _stprintf(szFreq, _T("%7.3f"),RadioPara.ActiveFrequency);
       int iIdx = SearchStation(Frequency);
       if(iIdx != 0)
       {
-	_stprintf(Name,_T("%s"),WayPointList[iIdx].Name);
-	_stprintf( RadioPara.ActiveName,_T("%s"),WayPointList[iIdx].Name);
-	ActiveRadioIndex = iIdx;
+    	 	CopyTruncateString(RadioPara.ActiveName, NAME_SIZE, WayPointList[iIdx].Name);
```
Comment (brunotl) [inline Common/Source/Dialogs/dlgRadioSettings.cpp:410]: useless,  RadioPara.ActiveName will be overwrited by devPutFreqActive line 414

File: Common/Source/Dialogs/dlgRadioSettings.cpp [hunk 12]
```diff
@@ -394,16 +407,14 @@ _stprintf(szFreq, _T("%7.3f"),RadioPara.ActiveFrequency);
...
+        ActiveRadioIndex = iIdx;
 
       }
       devPutFreqActive(Frequency,Name);
- 	  RadioPara.ActiveFrequency = Frequency;
-
+ 	    RadioPara.ActiveFrequency = Frequency;
```
Comment (brunotl) [inline Common/Source/Dialogs/dlgRadioSettings.cpp:415]: alredy done by devPutFreqActive line 414

File: Common/Source/Dialogs/dlgRadioSettings.cpp [hunk 13]
```diff
@@ -394,16 +407,14 @@ _stprintf(szFreq, _T("%7.3f"),RadioPara.ActiveFrequency);
...
       RadioPara.Changed =TRUE;
- //	 OnRemoteUpdate();
+
       OnUpdate();
     }
     OnUpdate();
```
Comment (brunotl) [inline Common/Source/Dialogs/dlgRadioSettings.cpp:420]: on Update called twice ?

File: Common/Source/Dialogs/dlgRadioSettings.cpp [hunk 14]
```diff
@@ -422,15 +433,14 @@ TCHAR	Name[NAME_SIZE] = _T("  ???   ");
...
-      if(iIdx != 0)
-      {
-	_stprintf(Name,_T("%s"),WayPointList[iIdx].Name);
-	_stprintf( RadioPara.PassiveName,_T("%s"),WayPointList[iIdx].Name);
-	PassiveRadioIndex = iIdx;
-      }
+     int iIdx = SearchStation(Frequency);
+     if(iIdx != 0)
+     {
+			 CopyTruncateString(RadioPara.PassiveName,NAME_SIZE, WayPointList[iIdx].Name);
```
Comment (brunotl) [inline Common/Source/Dialogs/dlgRadioSettings.cpp:439]: overwrited by line 442

File: Common/Source/Dialogs/dlgRadioSettings.cpp [hunk 15]
```diff
@@ -422,15 +433,14 @@ TCHAR	Name[NAME_SIZE] = _T("  ???   ");
...
+			 PassiveRadioIndex = iIdx;
+     }
      devPutFreqStandby(Frequency,Name);
-    RadioPara.PassiveFrequency = Frequency;
+     RadioPara.PassiveFrequency = Frequency;
```
Comment (brunotl) [inline Common/Source/Dialogs/dlgRadioSettings.cpp:443]: alredy done by line 442

File: Common/Source/Dialogs/dlgRadioSettings.cpp [hunk 16]
```diff
@@ -448,25 +458,33 @@ static void OnRadioActiveAutoClicked(WndButton* pWnd){
 		if ( ValidWayPoint(BestAlternate))
 		{
 			double fFreq = StrToDouble(WayPointList[BestAlternate].Freq,NULL);
-			if(ValidFrequency(fFreq))
-				devPutFreqActive(fFreq, WayPointList[BestAlternate].Name);
+
+			if(ValidFrequency(fFreq)) {
+				RadioPara.ActiveFrequency = fFreq;
+				CopyTruncateString(RadioPara.ActiveName, NAME_SIZE, WayPointList[BestAlternate].Name);
+				devPutFreqActive(	RadioPara.ActiveFrequency , RadioPara.ActiveName);
```
Comment (brunotl) [inline Common/Source/Dialogs/dlgRadioSettings.cpp:465]: this 3 line are same as 
    devPutFreqActive(fFreq, WayPointList[BestAlternate].Name);

File: Common/Source/Dialogs/dlgRadioSettings.cpp [hunk 17]
```diff
@@ -448,25 +458,33 @@ static void OnRadioActiveAutoClicked(WndButton* pWnd){
...
 		if ( ValidWayPoint(BestAlternate))
 		{
 			double fFreq = StrToDouble(WayPointList[BestAlternate].Freq,NULL);
-			if(ValidFrequency(fFreq))
-				devPutFreqStandby(fFreq, WayPointList[BestAlternate].Name);
+			if(ValidFrequency(fFreq))			{
+				RadioPara.PassiveFrequency = fFreq;
+				CopyTruncateString(RadioPara.PassiveName,NAME_SIZE,WayPointList[BestAlternate].Name);
+				devPutFreqStandby(	RadioPara.PassiveFrequency, RadioPara.PassiveName);
+			}
```
Comment (brunotl) [inline Common/Source/Dialogs/dlgRadioSettings.cpp:487]: same as previous

File: Common/Source/Dialogs/dlgRadioSettings.cpp (hunk 5 above)
Comment (AlphaLima) [reply Common/Source/Dialogs/dlgRadioSettings.cpp:190]: Some radios (KRT2) send an string full of spaces others an empty string

File: Common/Source/Dialogs/dlgRadioSettings.cpp (hunk 4 above)
Comment (AlphaLima) [reply Common/Source/Dialogs/dlgRadioSettings.cpp:165]: ok changed

File: Common/Source/Dialogs/dlgRadioSettings.cpp (hunk 8 above)
Comment (AlphaLima) [reply Common/Source/Dialogs/dlgRadioSettings.cpp:277]: local copy Name is shorter to add brackets lest and right 
[Stationname] = remote update
>Stationname< = Radio receiving
<Stationname> = Redio sending

File: Common/Source/Dialogs/dlgRadioSettings.cpp (hunk 13 above)
Comment (AlphaLima) [reply Common/Source/Dialogs/dlgRadioSettings.cpp:420]: removed

File: Common/Source/Dialogs/dlgRadioSettings.cpp (hunk 5 above)
Comment (brunotl) [reply Common/Source/Dialogs/dlgRadioSettings.cpp:190]: in this case name must be trimmed when it is received ...

File: Common/Source/Dialogs/dlgRadioSettings.cpp (hunk 8 above)
Comment (brunotl) [reply Common/Source/Dialogs/dlgRadioSettings.cpp:277]: this : 
          _stprintf(Name,_T("%s"),WayPointList[PassiveRadioIndex].Name);
          if(wpnewPassive)
            wpnewPassive->SetCaption(Name);

is same as
          if(wpnewPassive)
            wpnewPassive->SetCaption(WayPointList[PassiveRadioIndex].Name);
 
so copy is useless ...

**Pull request merged**